### ISSUE TITLE
Add fusion optimization for Delayed

### ISSUE THREAD

**Issue by gjoseph92** [dask/dask#8448]
We're starting to realize how critical the task fusion optimization is for avoiding task overproduction (see dask/distributed#5555, dask/distributed#5223 for background). 

In light of this and https://github.com/dask/dask/issues/8447, I think we should add the `fuse` optimization for `delayed` as well, which currently doesn't have it: https://github.com/dask/dask/blob/a5aecac8313fea30c5503f534c71f325b1775b9c/dask/delayed.py#L472-L478

I could see the counterargument that Delayed is lower-level, and users expect each of their delayed functions to run as a separate task. I'm not sure how much users want this, versus want help avoiding the very poor scheduling we currently have for unfused chains. We could always take the https://github.com/dask/dask/pull/7620 approach and have `optimization.fuse.active=None` fuse Delayed by default, but turn off fusion if explicitly set to False.

cc @rjzamora @ian-r-rose @jcrist @jrbourbeau

**Comment by jrbourbeau** [comment]
Thanks for the proposal @gjoseph92. For reference, @jcrist implemented this in https://github.com/dask/dask/pull/6222, but backed off due to other HLG work that was going on. That said, it's not totally clear what work he was referencing -- maybe @jcrist can clarify?

**Comment by martindurant** [comment]
> users expect each of their delayed functions to run as a separate task.

I don't think this is a strong argument, at least in the normal case that the tasks don't have extra annotations like retry.

**Comment by davidhao3300** [comment]
Was about to file an issue but discovered this one, here's some code that illustrates one possible motivation for having delayed fusion support:

Basically we want to do a computation but also get some intermediate results. We're also really trying to fuse tasks together into a single node where possible. Our thought was that we can "join" the results we want together in a parent computation node that just returns the results. This way dask thinks that the whole thing is one chain and will hopefully fuse together the whole thing

```python
import pandas as pd
import dask.dataframe as dd
import dask

dask.config.set({"optimization.fuse.active": True})
df = pd.DataFrame({'a': list(range(1000))})
ddf = dd.from_pandas(df, npartitions=1)


def f(x):
    return x
def g(x):
    return x
ddf2 = ddf.map_partitions(g, meta=ddf)
ddf3 = ddf.map_partitions(f, meta=ddf)

ddf = ddf2 + ddf3

joined_ddf = dask.delayed(f)([ddf, ddf2])

# Works as expected - one fused node
# dask.visualize(ddf, optimize_graph=True)

# Both are unoptimized
# This one is b/c the intermediate output is requested by the client
# dask.visualize(ddf, ddf2, optimize_graph=True)

# This one should be optimized and fused into a single node
dask.visualize(joined_ddf, optimize_graph=True)
```

This results in the following optimized graph:
<img width="218" alt="Screen Shot 2021-12-14 at 1 22 35 PM" src="https://user-images.githubusercontent.com/5534398/146082006-2f37d8a8-fb20-4cf8-ad65-2fc82630f285.png">

**Comment by gjoseph92** [comment]
FWIW this is probably much less critical for performance now that queuing is on by default, which fixes root task overproduction.
